checker/src/org/checkerframework/checker/index/upperbound/UpperBoundTransfer.java

```diff
@@ -46,8 +47,20 @@ public UpperBoundTransfer(CFAnalysis analysis) {
             AssignmentNode node, TransferInput<CFValue, CFStore> in) {
         TransferResult<CFValue, CFStore> result = super.visitAssignment(node, in);
 
-        if (node.getExpression() instanceof ArrayCreationNode) {
-            ArrayCreationNode acNode = (ArrayCreationNode) node.getExpression();
+        Node expNode = node.getExpression();
+
+        ArrayCreationNode acNode =
+                expNode instanceof ArrayCreationNode ? (ArrayCreationNode) expNode : null;
+
+        if (expNode instanceof TypeCastNode) {
```

Comment:
Shouldn't this be a `while` loop rather than an `if` statement?  And shouldn't it also call `TreeUtils.skipParens`?  The `while`  loop could be patterned after that one.  Or there could be a new routine that skips over parens and typecasts.